src/features/script-load.js

```diff
@@ -12,12 +12,15 @@ systemJSPrototype.register = function (deps, declare) {
 systemJSPrototype.instantiate = function (url, firstParentUrl) {
   const loader = this;
   if (url.substr(-5) === '.json') {
-    return fetch(url).then(function (resp) {
-      return resp.text();
-    }).then(function (source) {
-      return [[], function(_export) {
-        return {execute: function() {_export('default', JSON.parse(source))}};
-      }];
+    return loadDynamicModule(url, function (_export, source) {
+      _export('default', JSON.parse(source));
+    });
+  } else if (url.substr(-4) === '.css') {
```

Comment:
Modules types are determined by the response mime type, not the request extension. Given that this is a fetch, can you check the response headers here?

Reply:
SystemJS has to choose upfront whether to do a fetch request or append a script element. Fetch is for json/css/html modules and script for js modules. Script element for js modules is desireable over fetch + eval for a variety of reasons, including performance and sourcemaps.

Another thing to take note of is that it is not possible to check the response headers, including mime type, for script elements.

For JSON modules, we decided to use the file extension as a best-effort implementation since we couldn't think of a way to overcome the challenges described above ^

My implementation here of CSS modules has the same challenges and I went with the same best-effort implementation that is based on file extension.

Reply:
I just pushed a commit that adds a "Limitations" section that explains this deviation from the spec in the docs.